library/alloc/src/rc.rs

```diff
@@ -386,7 +386,35 @@ impl<T> Rc<T> {
     /// returns, calling [`upgrade`] on the weak reference inside `data_fn` will
     /// fail and result in a `None` value.
     ///
+    /// # Why using this method?
+    ///
+    /// It allows to construct a `T` which holds a `Weak<T>` pointer to itself.
+    /// Without this method, you would need to use [`Weak::new`] and then
+    /// mutate the value by using [`Rc::get_mut`] to fix up the self references.
+    ///
+    /// # Why is a closure needed?
+    ///
+    /// The place needs to be "finished" once the value is created. In theory,
+    /// you could "recover" any zombie weak, but for the most case, Rust wants
```

Comment:
The term "zombie weak" is introduced here. While I can guess what it might mean, it's not defined anywhere. Maybe we should just avoid the term.